Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAssetHandler.cpp

```diff
@@ -36,14 +36,155 @@ namespace AZ
                 if (loadResult == Data::AssetHandler::LoadResult::LoadComplete)
                 {
                     // ImageMipChainAsset has some internal variables need to initialized after it was loaded.
-                    StreamingImageAsset* assetData2 = asset.GetAs<StreamingImageAsset>();
-                    assetData2->m_tailMipChain.Init();
+                    assetData->m_tailMipChain.Init();
+
+                    // Handle StreamingImageAsset reload (which the asset can be found from asset manager)
+                    auto& assetManager = AZ::Data::AssetManager::Instance();
+                    Data::Asset<StreamingImageAsset> foundImageAsset = assetManager.FindAsset(asset.GetId(), AZ::Data::AssetLoadBehavior::PreLoad);
```

Comment:
Nit: Usually prefer to use the Default load behavior, it doesn't affect the code any since these references aren't saved, but it looks less confusing to use the default, as it doesn't give people the impression that it's going to affect the load in some way.